cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java

```diff
@@ -178,28 +178,22 @@ public AdapterSpecification getAdapter(String namespace, String adapterName) thr
    * @throws IllegalArgumentException on errors.
    */
   public void createAdapter(String namespaceId, AdapterSpecification adapterSpec) throws IllegalArgumentException {
-
     AdapterTypeInfo adapterTypeInfo = adapterTypeInfos.get(adapterSpec.getType());
     Preconditions.checkNotNull(adapterTypeInfo, "Adapter type %s not found", adapterSpec.getType());
 
-    try {
-      ApplicationSpecification appSpec = deployApplication(namespaceId, adapterTypeInfo);
-
-      validateSources(adapterSpec.getName(), adapterSpec.getSources());
-      createSinks(adapterSpec.getSinks(), adapterTypeInfo);
+    ApplicationSpecification appSpec = deployApplication(namespaceId, adapterTypeInfo);
 
-      // If the adapter already exists, remove existing schedule to replace with the new one.
-      AdapterSpecification existingSpec = store.getAdapter(Id.Namespace.from(namespaceId), adapterSpec.getName());
-      if (existingSpec != null) {
-        unschedule(namespaceId, appSpec, adapterTypeInfo, existingSpec);
-      }
-
-      schedule(namespaceId, appSpec, adapterTypeInfo, adapterSpec);
-      store.addAdapter(Id.Namespace.from(namespaceId), adapterSpec);
+    validateSources(adapterSpec.getName(), adapterSpec.getSources());
+    createSinks(adapterSpec.getSinks(), adapterTypeInfo);
 
-    } catch (Exception e) {
-      throw new RuntimeException("Error deploying adapter application " + e.getMessage());
+    // If the adapter already exists, remove existing schedule to replace with the new one.
+    AdapterSpecification existingSpec = store.getAdapter(Id.Namespace.from(namespaceId), adapterSpec.getName());
+    if (existingSpec != null) {
+      unschedule(namespaceId, appSpec, adapterTypeInfo, existingSpec);
```

Comment:
I'm changing this in another PR (#1099). Already implemented, just waiting for this to be merged, as that PR is branched off here.
(keeping the PRs smaller).